src/input.rs

```diff
@@ -69,10 +71,24 @@ impl<R: Read> Iterator for Events<R> {
             Err(e) => Err(e),
         };
 
-        Some(res.or(Ok(Event::Unsupported)))
+        Some(res)
     }
 }
 
+fn parse_event<I>(item: u8, iter: &mut I) -> Result<Event, io::Error>
```

Comment:
Wrap `event::parse_event`, saving any sent byte in a buffer.
This is an easier way to report an unsupported sequence.